back/infolica/views/reference_numeros.py

```diff
@@ -27,20 +27,41 @@ def reference_numeros_new_view(request):
     numeros_liste = json.loads(
         request.params['numeros_liste']) if 'numeros_liste' in request.params else None
 
+    # Get affaire_type_id and config values
+    affaire_type_id = request.dbsession.query(Affaire).filter(Affaire.id == affaire_id).first().type_id
+    affaire_type_cadastration_id = int(request.registry.settings['affaire_type_cadastration_id'])
+    facture_type_facture_id = int(request.registry.settings['facture_type_facture_id'])
+    client_cadastration_id = int(request.registry.settings['client_cadastration_id'])
+
     for numero_i in numeros_liste:
         # enregistrer le lien affaire-numéro
         params = Utils._params(
             affaire_id=affaire_id, numero_id=numero_i['numero_id'], actif=True, type_id=1)
         affaire_numero_new_view(request, params)
 
+        # Si l'affaire est une cadastration, générer une facture par numéro
+        if affaire_type_id == affaire_type_cadastration_id:
+            params = Utils._params(
+              type_id = facture_type_facture_id,
+              affaire_id = affaire_id,
+              client_id = client_cadastration_id,
+              montant_mat_diff = 0,
+              montant_mo = 0,
+              montant_rf = 0,
+              montant_total = 0,
+              montant_tva = 0,
+              numeros = [numero_i['numero_id']]
```

Comment:
```suggestion
              type_id=facture_type_facture_id,
              affaire_id=affaire_id,
              client_id=client_cadastration_id,
              montant_mat_diff=0,
              montant_mo=0,
              montant_rf=0,
              montant_total=0,
              montant_tva=0,
              numeros=[numero_i['numero_id']]
```

Ce sont des paramètres de fonction, pas d'espace dans Python